f3probe.c

```diff
@@ -485,6 +486,7 @@ static int test_device(struct args *args)
 	case FKTY_CHAIN: {
 		last_good_sector = (real_size_byte >> 9) - 1;
 		assert(block_order >= 9);
+		assert(asprintf(&fix_cmd, "f3fix --last-sec=%" PRIu64 " %s", last_good_sector, final_dev_filename) > 0);
 		printf("Bad news: The device `%s' is a counterfeit of type %s\n\n"
 			"You can \"fix\" this device using the following command:\n"
 			"f3fix --last-sec=%" PRIu64 " %s\n",
```

Comment:
1. Replace the line `"f3fix --last-sec=%" PRIu64 " %s\n",` with `"%s\n",`.
2. Replace the line `last_good_sector, final_dev_filename);` below with `fix_cmd);`.